networkx/generators/degree_seq.py

```diff
@@ -348,43 +427,41 @@ def expected_degree_graph(w, seed=None, selfloops=True):
        pp. 115-126, 2011.
     """
     n = len(w)
-    G=nx.empty_graph(n)
-    if n==0 or max(w)==0: # done if no edges
+    G = nx.empty_graph(n)
+
+    # If there are no nodes are no edges in the graph, return the empty graph.
+    if n == 0 or max(w) == 0:
         return G
+
     if seed is not None:
         random.seed(seed)
-    rho = 1/float(sum(w))
-    # sort weights, largest first
-    # preserve order of weights for integer node label mapping
-    order = sorted(enumerate(w),key=itemgetter(1),reverse=True)
```

Comment:
I think as I stated in the comment below, the index of the weights corresponds to the node label. I think the original code is just doing a lot more work than necessary.